src/main/java/ch/njol/skript/expressions/ExprTarget.java

```diff
@@ -90,57 +95,67 @@ public Entity convert(LivingEntity en) {
 		});
 	}
 
-	@Override
-	public Class<? extends Entity> getReturnType() {
-		return type != null ? type.getType() : Entity.class;
-	}
-
-	@Override
-	public String toString(@Nullable Event e, boolean debug) {
-		if (e == null)
-			return "the target" + (type == null ? "" : "ed " + type) + (getExpr().isDefault() ? "" : " of " + getExpr().toString(e, debug));
-		return Classes.getDebugMessage(getAll(e));
-	}
-
-	@Override
-	public boolean setTime(int time) {
-		return super.setTime(time, EntityTargetEvent.class, getExpr());
-	}
-
 	@Override
 	@Nullable
 	public Class<?>[] acceptChange(ChangeMode mode) {
-		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE)
-			return CollectionUtils.array(LivingEntity.class);
+		switch (mode) {
+			case SET:
+			case RESET:
+			case DELETE:
+				return CollectionUtils.array(LivingEntity.class);
+		}
 		return super.acceptChange(mode);
 	}
 
 	@Override
 	public void change(Event e, @Nullable Object[] delta, ChangeMode mode) {
-		if (mode == ChangeMode.SET || mode == ChangeMode.DELETE) {
+		if (mode == ChangeMode.SET || mode == ChangeMode.RESET || mode == ChangeMode.DELETE) {// null will make the entity target-less (reset target) but for players it will remove them
 			LivingEntity target = delta == null ? null : (LivingEntity) delta[0];
 			for (LivingEntity entity : getExpr().getArray(e)) {
-				if (getTime() >= 0 && e instanceof EntityTargetEvent && entity.equals(((EntityTargetEvent) e).getEntity()) && !Delay.isDelayed(e)) {
+				if (e instanceof EntityTargetEvent && entity.equals(((EntityTargetEvent) e).getEntity()) && !Delay.isDelayed(e)) {
 					((EntityTargetEvent) e).setTarget(target);
-				} else if (entity instanceof Mob) {
-					((Mob) entity).setTarget(target);
+				} else {
+					if (entity instanceof Mob) {
+						((Mob) entity).setTarget(target);
+					} else if (entity instanceof Player && mode == ChangeMode.DELETE) {
+						Entity playerTarget = getTarget(entity, type);
+						if (playerTarget != null && !(playerTarget instanceof OfflinePlayer))
+							playerTarget.remove();
+					}
 				}
 			}
-			return;
+		} else {
+			super.change(e, delta, mode);
 		}
-		super.change(e, delta, mode);
```

Comment:
I like the old way of `return` rather than an else statement. Probably an oversight since you changed from switch to if statement, but I think we should keep the return and fall through rather than a single line else statement with braces.